lib/api/v3/render/render_api.rb

```diff
@@ -94,36 +103,23 @@ def parse_context
                 context
               end
             end
-
-            def renderer(type)
-              case type
-              when :textile
-                ::API::Utilities::Renderer::TextileRenderer.new(request_body, context_object)
-              when :plain
-                ::API::Utilities::Renderer::PlainRenderer.new(request_body)
-              end
-            end
-
-            def render(type)
-              renderer(type).to_html
-            end
           end
 
-          resources :textile do
-            post do
-              check_content_type
-              setup_response
-
-              render :textile
+          route_param :render_format do
+            before do
+              @format = params[:render_format]
+              @format = '' if @format == 'plain' # 'plain' is called '' internally
```

Comment:
@NobodysNightmare we talked about only using 'plain' from the outside an burying the logic within actual text formatter.